src/voter_set.rs

```diff
@@ -57,10 +55,9 @@ impl<Id: Eq + Ord> VoterSet<Id> {
 		Id: Ord + Clone,
 		I: IntoIterator<Item = (Id, u64)>,
 	{
-		let weights = weights.into_iter();
```

Comment:
it's an into_iter already isn't it?

Reply:
`weights` is something that can be converted into an iterator but we need to call `into_iter()` on it to get an iterator, I think the code won't compile if you remove this line.

Reply:
Nevermind, it should work now since you're not calling `size_hint` anymore and just using it in the for loop below.